src/java/org/apache/cassandra/config/DurationSpec.java

```diff
@@ -24,192 +24,119 @@
-    }
+        this (value, minUnit);
 
-    public TimeUnit unit()
-    {
-        return unit;
+        validateMinUnit(unit, minUnit, value);
+        validateQuantity(value, quantity(), unit(), minUnit, max);
     }
 
-    /**
-     * Creates a {@code DurationSpec} of the specified amount of milliseconds.
-     *
-     * @param milliseconds the amount of milliseconds
-     * @return a duration
-     */
-    public static DurationSpec inMilliseconds(long milliseconds)
+    private void validateMinUnit(TimeUnit unit, TimeUnit minUnit, String value)
     {
-        return new DurationSpec(milliseconds, MILLISECONDS);
+        if (unit.compareTo(minUnit) < 0)
+            throw new ConfigurationException(String.format("Invalid duration: %s Accepted units:%s", value, acceptedUnits(minUnit)));
     }
 
-    public static DurationSpec inDoubleMilliseconds(double milliseconds)
+    private String acceptedUnits(TimeUnit minUnit)
     {
-        return new DurationSpec(milliseconds, MILLISECONDS);
+        TimeUnit[] units = TimeUnit.values();
+        return Arrays.toString(Arrays.copyOfRange(units, minUnit.ordinal(), units.length));
     }
 
-    /**
-     * Creates a {@code DurationSpec} of the specified amount of seconds.
-     *
-     * @param seconds the amount of seconds
-     * @return a duration
-     */
-    public static DurationSpec inSeconds(long seconds)
+    private static void validateQuantity(String value, long quantity, TimeUnit sourceUnit, TimeUnit minUnit, long max)
     {
-        return new DurationSpec(seconds, SECONDS);
-    }
+        if (quantity < 0)
+            throw new ConfigurationException("Invalid duration: value must be non-negative");
 
-    /**
-     * Creates a {@code DurationSpec} of the specified amount of minutes.
-     *
-     * @param minutes the amount of minutes
-     * @return a duration
-     */
-    public static DurationSpec inMinutes(long minutes)
-    {
-        return new DurationSpec(minutes, MINUTES);
+        if (minUnit.convert(quantity, sourceUnit) >= max)
+            throw new ConfigurationException("Invalid duration: " + value + ". It shouldn't be more than " +
+                                             (max - 1) + " in " + minUnit.name().toLowerCase());
     }
 
-    /**
-     * Creates a {@code DurationSpec} of the specified amount of hours.
-     *
-     * @param hours the amount of hours
-     * @return a duration
-     */
-    public static DurationSpec inHours(long hours)
+    private static void validateQuantity(long quantity, TimeUnit sourceUnit, TimeUnit minUnit, long max)
     {
-        return new DurationSpec(hours, HOURS);
+        if (quantity < 0)
+            throw new ConfigurationException("Invalid duration: value must be non-negative");
+
+        if (minUnit.convert(quantity, sourceUnit) >= max)
+            throw new ConfigurationException("Invalid duration: " + quantity + " " + sourceUnit.name().toLowerCase() + ". It shouldn't be more than " +
```

Comment:
Perhaps this message is long enough to use `String.format` to increase readability:
```suggestion
            throw new ConfigurationException(String.format("Invalid duration: %d %s. It shouldn't be more than %d in %s",
                                                           quantity, sourceUnit.name().toLowerCase(),
                                                           max - 1, minUnit.name().toLowerCase()));
```